src/main/java/com/srans/nestserver/controller/PaymentController.java

```diff
@@ -39,56 +41,133 @@
 @RequestMapping("/api/v1")
 public class PaymentController {
 
-	
 	private Logger logger = LoggerFactory.getLogger(PaymentController.class);
-	
-
 
 	@Autowired
 	private PaymentRepository paymentRepository;
- 
+
 	@Autowired
-	private HostelRepository hostelRepository; 
+	private HostelRepository hostelRepository;
 
 	@Autowired
 	private UserRepository userRepository;
- 
-	@GetMapping("/payments") 
-	@PreAuthorize("permitAll()") 
-	public List<Payment> getPaymentHistory(@RequestParam(value = "id") Long userId, @RequestParam(value = "role") String role) throws ResourceNotFoundException {
-		
-		List<Payment> payments = null;
-		logger.info("IN::GET::/payments::getPaymentHistory::" + userId+"::"+role);
-		if(role != null && role.equalsIgnoreCase(NSConstants.ROLE_TENANT)) {
-			payments = paymentRepository.getPaymentHistory(userId);
-		}else {
-			payments = paymentRepository.getAllUserPaymentsForAdmin(userId);
+
+	@Autowired
+	private RoomRepository roomRepository;
+
+	@GetMapping("/payments")
+	@PreAuthorize("permitAll()")
+	public List<PaymentHistory> getPaymentHistory(@RequestParam(value = "id") Long userId,
+			@RequestParam(value = "role") String role) throws ResourceNotFoundException {
+
+		// List<Payment> payments = null;
+		logger.info("IN::GET::/payments::getPaymentHistory::" + userId + "::" + role);
+		List<PaymentHistory> paymentList = new ArrayList<>();
+
+		PaymentHistory payment = new PaymentHistory();
+		if (role != null && role.equalsIgnoreCase(NSConstants.ROLE_TENANT)) {
+
+			paymentRepository.getPaymentHistory(userId).stream().forEach(payments -> {
+
+				if (payments.getUserId() == userId) {
+					payment.setId(payments.getId());
```

Comment:
Why this if conditions again, u already queries with userId meaning only those the records matching corresponding userId will be returned from that caller, this condition not required u can remove that